docs/docs/integrations/retrievers/azure_ai_search.ipynb

```diff
@@ -0,0 +1,292 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "1edb9e6b",
+   "metadata": {},
+   "source": [
+    "# Azure AI Search\n",
+    "\n",
+    ">[Azure AI Search](https://learn.microsoft.com/en-us/azure/search/search-what-is-azure-search) (formerly known as `Azure Cognitive Search`or Azure Search) is a cloud search service that gives developers infrastructure, APIs, and tools for building a rich search experience over private, heterogeneous content in web, mobile, and enterprise applications.\n",
+    "\n",
+    ">Search is foundational to any app that surfaces text to users, where common scenarios include catalog or document search, online retail apps, or data exploration over proprietary content. When you create a search service, you'll work with the following capabilities:\n",
+    "> - A search engine for full text search over a search index containing user-owned content\n",
+    "> - Rich indexing, with lexical analysis and optional AI enrichment for content extraction and transformation\n",
+    "> - Rich query syntax for text search, fuzzy search, autocomplete, geo-search and more\n",
+    "> - Programmability through REST APIs and client libraries in Azure SDKs\n",
+    "> - Azure integration at the data layer, machine learning layer, and AI (AI Services)\n",
+    "\n",
+    "This notebook shows how to use Azure AI Search (AAS) within LangChain."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "074b0004",
+   "metadata": {},
+   "source": [
+    "## Set up Azure AI Search\n",
+    "\n",
+    "To set up AAS, please follow the instructions [here](https://learn.microsoft.com/en-us/azure/search/search-create-service-portal).\n",
+    "\n",
+    "Please note you will need:\n",
+    "\n",
+    "1. the name of your AAS service, \n",
+    "2. the name of your AAS index,\n",
+    "3. your API key.\n",
```

Comment:
Were you able to add support for role based authentication?

https://learn.microsoft.com/en-us/azure/search/search-security-rbac?tabs=config-svc-portal%2Croles-portal%2Ctest-portal%2Ccustom-role-portal%2Cdisable-keys-portal

Reply:
I haven't yet! I'll open a separate PR for this!